bootstrap.sh

```diff
@@ -49,6 +49,18 @@ program_exists() {
     local ret='0'
     type $1 >/dev/null 2>&1 || { local ret='1'; }
 
+    # fail on non-zero return value
+    if [ ! "$ret" -eq '0' ]; then
+        return 1
+    fi
+
+    return 0
+}
+
+program_must_exist() {
+    local ret='0'
+    type $1 >/dev/null 2>&1 || { local ret='1'; }
```

Comment:
Couldn't we just re-use the `program_exists` function here and then call `error` if we get 1?